### PR TITLE
Refs [TASK][35689] Unit Test Movie Detail

### PR DESCRIPTION
**Pull request by khoaqh-2470**
## Related Tickets
- [#Task 35689: Unit Test Movie Detail and FixBugs](https://edu-redmine.sun-asterisk.vn/issues/35689)
## WHAT
- Screenshot from 2021-04-22 13-56-03
## Review Checklist

Category | View Point | Description | Self review | Reviewer2 (name)
--- | --- | --- | --- | ---
Conventions | Sun* coding conventions | https://github.com/framgia/coding-standards/tree/master/eng/android |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Conventions | Kotlin coding conventions | https://kotlinlang.org/docs/coding-conventions.html |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Redmine | Sun* Redmine working process  | https://github.com/framgia/Training-Guideline/blob/master/WorkingProcess/redmine/redmine.md |<li>- [ ] yes</li>|<li>- [ ] yes</li>  

## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(List gem, library third party add new)*

*(Other notes)*

### CODE REVIEW

File: app/src/test/java/com/sun/moviedb_53/ui/detail/movie/MovieDetailPresenterTest.kt [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+    private lateinit var detail: MovieDetail
+    private lateinit var listActor: List<Actor>
+    private lateinit var listRecommendation: List<HotMovie>
+
+    @Before
+    fun setUp() {
+        presenter = MovieDetailPresenter(repository, favoriteRepository)
+        presenter.setView(view)
+        detail =
+            MovieDetail(0, "abc", "abc", 0.0, "abc", "abc", mutableListOf(), "abc", "abc")
```
Comment (ghost) [inline app/src/test/java/com/sun/moviedb_53/ui/detail/movie/MovieDetailPresenterTest.kt]: Create file FakeData.kt

const val   MOVIE_DETAIL = MovieDetail(0, "abc", "abc", 0.0, "abc", "abc", mutableListOf(), "abc", "abc")

File: app/src/test/java/com/sun/moviedb_53/ui/detail/movie/MovieDetailPresenterTest.kt [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+            repository.getMovieDetails(
+                ArgumentMatchers.anyInt(),
+                any()
+            )
+        ).thenAnswer {
+            (it.arguments[1] as OnFetchDataJsonListener<MovieDetail>).onSuccess(
+                detail
+            )
+        }
+        presenter.getMovieDetail(399566)
```
Comment (ghost) [inline app/src/test/java/com/sun/moviedb_53/ui/detail/movie/MovieDetailPresenterTest.kt]: Move 399566 to FakeData.kt

File: app/src/test/java/com/sun/moviedb_53/ui/detail/movie/MovieDetailPresenterTest.kt (hunk 1 above)
Comment (khoaqh-2470) [reply app/src/test/java/com/sun/moviedb_53/ui/detail/movie/MovieDetailPresenterTest.kt]: Anh ơi. Nó k hiểu const á, dùng như này mới được
val MOVIE_DETAIL =
        MovieDetail(0, "abc", "abc", 0.0, "abc", "abc", mutableListOf(), "abc", "abc")
này thôi anh.
Dùng const nó lỗi á.
const nó dùng cho chuỗi với kiểu nguyên thủy thôi á a

Comment (ghost) [reply app/src/test/java/com/sun/moviedb_53/ui/detail/movie/MovieDetailPresenterTest.kt]: dùng val đi lỗi tí :v:

**Review by ghost — APPROVED** [review]

**Review by tuanlvt-1117 — APPROVED** [review]

**Pull request merged**